### PR TITLE
Notification patches

### PR DESCRIPTION
**Pull request by pandurx**

### CODE REVIEW

File: mod/cp_notifications/lib/functions.php [hunk 1]
```diff
@@ -767,10 +748,9 @@ function render_contents($content_array, $heading = '', $language_preference = '
 
 	    } else {
 
-	   	  $author = $content_array['content_author'];
-
+	   	  	$author = $content_array['content_author'];
 
-			$content_title = (is_array($content_array['content_title'])) ? $content_array['content_title'][$language_preference] : $content_title = $content_array['content_title'];
+	   	  	$content_title = gc_explode_translation($content_array['content_title'], $language_preference);
```
Comment (nschonni) [inline mod/cp_notifications/lib/functions.php]: Looks like this got mixed tabs/spaces

File: mod/cp_notifications/start.php [hunk 2]
```diff
@@ -1532,8 +1532,8 @@ function cp_digest_daily_cron_handler($hook, $entity_type, $return_value, $param
 
 
 			// delete and clean up the notification, already sent so we don't need to keep it anymore
-			$query = "DELETE FROM notification_digest WHERE user_guid = {$user->getGUID()}";
-			$result = delete_data($query);
+			//$query = "DELETE FROM notification_digest WHERE user_guid = {$user->getGUID()}";
```
Comment (nschonni) [inline mod/cp_notifications/start.php]: Delete me?

Comment (pandurx) [reply mod/cp_notifications/start.php]: I commented that out so I can do testing (i had forgotten to uncomment when I did the commit)

Comment (nschonni) [reply mod/cp_notifications/start.php]: Np, you can use this https://help.github.com/articles/changing-a-commit-message/ to just fixup the last commit. You need to do a `git push -f` after though so make sure the push settings are correct (or you're using protected branches in the repo for safety)

File: mod/cp_notifications/start.php [hunk 3]
```diff
@@ -905,13 +918,15 @@ function cp_create_annotation_notification($event, $type, $object) {
 		foreach ($to_recipients_site as $to_recipient_id => $to_recipient) {
 			$site_template = elgg_view('cp_notifications/site_template', $message);
 			$recipient_user = get_user($to_recipient->guid);
-			
-			if (strcmp(elgg_get_plugin_user_setting('cpn_set_digest', $to_recipient->guid,'cp_notifications'),'set_digest_yes') !== 0) {
 
-				if (has_access_to_entity($object, $recipient_user)) {
+			if ($object->access_id == 1 || $object->access_id == 2 || $content_entity->getType() === 'group' || $action_type === 'post_likes') {
+
+				if (strcmp(elgg_get_plugin_user_setting('cpn_set_digest', $to_recipient->guid,'cp_notifications'),'set_digest_yes') !== 0) {
```
Comment (nschonni) [inline mod/cp_notifications/start.php]: Could probably merge these new condtions with the outer `if`

File: mod/cp_notifications/views/default/cp_notifications/email_template.php [hunk 4]
```diff
@@ -432,17 +432,23 @@
...
 			$cp_notify_msg_description_fr = elgg_echo('cp_personalized_message',array($vars['cp_email_invited_by']->name,$vars['cp_invitation_msg']),'fr');
 		}
 
-		$cp_notify_msg_description_en .= elgg_echo('cp_notify:body_group_invite_email:description',array($vars['cp_invitation_non_user_url'],'http://www.gcpedia.gc.ca/wiki/GCconnex_User_Help/Content_Management_and_Collaboration/How_Do_I_Access_and_Join_a_Group%3F'.'?utm_source=notification&utm_medium=email',$vars['cp_invitation_code']),'en');
-		$cp_notify_msg_description_fr .= elgg_echo('cp_notify:body_group_invite_email:description',array($vars['cp_invitation_non_user_url'],'http://www.gcpedia.gc.ca/wiki/GCconnex_-_Aide_%C3%A0_l%27utilisateur/Pour_commencer/Comment_puis-je_acc%C3%A9der_%C3%A0_un_groupe_et_m%E2%80%99y_joindre%3F'.'?utm_source=notification&utm_medium=email',$vars['cp_invitation_code']),'fr');
+		$help_url_link_en = "http://www.gcpedia.gc.ca/wiki/GCconnex_User_Help/Content_Management_and_Collaboration/How_Do_I_Access_and_Join_a_Group%3F?utm_source=notification&utm_medium=email";
```
Comment (nschonni) [inline mod/cp_notifications/views/default/cp_notifications/email_template.php]: This was there before, but maybe this URL should be in the translation strings?

File: mod/gc_tags/actions/blog/save.php [hunk 5]
```diff
@@ -1,193 +1,193 @@
...
-		}
-	} else {
-		register_error(elgg_echo('blog:error:cannot_save'));
-		forward($error_forward_url);
-	}
-} else {
-	register_error($error);
-	forward($error_forward_url);
-}
+<?php
```
Comment (nschonni) [inline mod/gc_tags/actions/blog/save.php]: For some reason this file is getting marked changed. Did you fix bad line endings or something?

File: mod/cp_notifications/lib/functions.php [hunk 6]
```diff
@@ -767,10 +748,9 @@ function render_contents($content_array, $heading = '', $language_preference = '
 
 	    } else {
 
-	   	  $author = $content_array['content_author'];
+	   	  	$author = $content_array['content_author'];
```
[the remark this answers is not in the document]
Comment (pandurx) [reply mod/cp_notifications/lib/functions.php]: as it looks fine in my editor, i'll leave it as is

File: mod/gc_tags/actions/blog/save.php (hunk 5 above)
Comment (pandurx) [reply mod/gc_tags/actions/blog/save.php]: i think i just deleted everything and then did an undo (i needed to figure out which blog/save it was using), it shouldn't pick up on that change though

File: mod/cp_notifications/lib/functions.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (nschonni) [reply mod/cp_notifications/lib/functions.php]: I think I said you were using Sublime. I used to use this https://github.com/disq/HighlightWhitespaces along with https://github.com/sindresorhus/editorconfig-sublime#readme to see/fix those issues

**Pull request merged**